sympy/stats/crv.py

```diff
@@ -258,7 +258,9 @@ def compute_entropy(self, **kwargs):
         """
         x = symbols('x', real = True, cls = Dummy)
         pdf = self.pdf(x)
-        h = integrate(-pdf * log(pdf), (x, -oo, oo))
+        lower_bound = self.set.start
+        upper_bound = self.set.end
+        h = integrate(-pdf * log(pdf), (x, lower_bound, upper_bound))
```

Comment:
Would it be possible to use something like `self.expectation` or `self.compute_expectation` that would automatically set the proper integration limits?

Reply:
I don't think this should be done, since it will give too complex results even for simple answers. Instead, to avoid repetition of code, we can implement a method `stats_integrate` which can be used in computing, `expectation`, `entropy`, `mgf`, `cf` alike. It will automatically set the domain.

Reply:
In which way would the results be more complicated?

Reply:
Using compute_expectation works [here](https://github.com/sympy/sympy/pull/15693/files#diff-6f227c2ec704bbb00895cfa85e7f4d03R557) and gives results as expected. However, it in the above method was giving complex and unevaluated Integrals. Thanks for the suggestion. I will update the PR and apologies for commenting without trying the `compute_expectation` as I mentioned above. :-)